jina/orchestrate/flow/base.py

```diff
@@ -2369,26 +2382,113 @@ def block(
+                ):
+                    from watchfiles import watch
+
+                new_stop_event = stop_event or threading.Event()
+                if len(watch_files_list) > 0:
                     while True:
-                        if self._stop_event.is_set():
+                        for changes in watch(
+                            *watch_files_list, stop_event=new_stop_event
+                        ):
+                            chanded_files = [
+                                changed_file for _, changed_file in changes
+                            ]
+                            for changed_file in chanded_files:
+                                if changed_file not in watch_files_from_deployments:
+                                    # maybe changed_file is the absolute path of one in watch_files_from_deployments
+                                    is_absolute_path = False
+                                    for (
+                                        file,
+                                        deployment_name,
+                                    ) in watch_files_from_deployments.items():
+                                        if changed_file.endswith(file):
+                                            is_absolute_path = True
+                                            _restart_deployment(
+                                                self._deployment_nodes[deployment_name],
+                                                changed_file,
+                                            )
+                                            break
+
+                                    if not is_absolute_path:
+                                        _restart_flow(
+                                            [
+                                                changed_file
+                                                for _, changed_file in changes
+                                            ][0]
+                                        )
+                                else:
+                                    _restart_deployment(
+                                        self._deployment_nodes[
+                                            watch_files_from_deployments[changed_file]
+                                        ],
+                                        changed_file,
+                                    )
+                        if new_stop_event.is_set():
                             break
-                        time.sleep(0.5)
             else:
-                if not __windows__:
-                    stop_event.wait()
+                if not stop_event:
+                    self._stop_event = threading.Event()
+                    if not __windows__:
+                        self._stop_event.wait()
+                    else:
+                        while True:
+                            if self._stop_event.is_set():
+                                break
+                            time.sleep(0.5)
```

Comment:
```suggestion
                    stop_event = threading.Event()
```

basically the next changes are about setting the new event object and then keeping the same stop_event wait logic common

Reply:
no, it needs to be set to self because the context manager of Flow requires it